pinot-core/src/test/java/org/apache/pinot/core/realtime/impl/fakestream/FakePartitionLevelConsumer.java

```diff
@@ -114,6 +114,10 @@ public MessageBatch fetchMessages(StreamPartitionMsgOffset startOffset, StreamPa
     }
     int startOffsetInt = (int) ((LongMsgOffset) startOffset).getOffset();
     int endOffsetInt = (int) ((LongMsgOffset) endOffset).getOffset();
+    if (endOffsetInt > _messageOffsets.size()) {
+      // Hack to get multiple batches
```

Comment:
Should we control this on the caller side? Message fetching should not be aware of the flush threshold